core/src/main/java/org/kohsuke/stapler/export/Property.java

```diff
@@ -199,118 +198,127 @@ private void writeValue(Type expected, Object value, TreePruner pruner, DataWrit
 
         Class c = value.getClass();
 
-        Model model;
         try {
-            model = owner.get(c, parent.type, name);
-        } catch (NotExportableException ex) {
-            if(STRING_TYPES.contains(c)) {
-                writer.value(value.toString());
-                return;
-            }
-            if(PRIMITIVE_TYPES.contains(c)) {
-                writer.valuePrimitive(value);
-                return;
-            }
-            Class act = c.getComponentType();
-            if (act !=null) { // array
-                Range r = pruner.getRange();
-                writer.startArray();
-                if (value instanceof Object[]) {
-                    // typical case
-                    for (Object item : r.apply((Object[]) value)) {
-                        writeBuffered(act, item, pruner, writer);
-                    }
-                } else {
-                    // more generic case
-                    int len = Math.min(r.max, Array.getLength(value));
-                    for (int i=r.min; i<len; i++) {
-                        writeBuffered(act, Array.get(value, i), pruner, writer);
-                    }
+            writer.type(expected, value.getClass());
```

Comment:
It should move down inside else condition below, like 211. handleNotExportable has its own logic to write type.